docs/docs/en/guides/develop/build/registry.md

```diff
@@ -0,0 +1,109 @@
+
+If you are using the <LocalizedLink href="/guides/develop/projects/dependencies#tuists-xcodeprojbased-integration">XcodeProj-based integration</LocalizedLink>, you can use the ``--replace-scm-with-registry`` flag to resolve dependencies from the registry if they are available. Add it to the `installOptions` in your `Tuist.swift` file:
+```swift
+import ProjectDescription
+
+let tuist = Tuist(
+    fullHandle: "{account-handle}/{project-handle}",
+    project: .tuist(
+        installOptions: .options(passthroughSwiftPackageManagerArguments: ["--replace-scm-with-registry"])
+    )
+)
+```
+
+If you want to ensure that the registry is used every time you resolve dependencies, you will need to update `dependencies` in your `Tuist/Package.swift` file to use the registry identifier instead of a URL. The registry identifier is always in the form of `{organization}.{repository}`. For example, to use the registry for the `swift-composable-architecture` package, do the following:
+```diff
+dependencies: [
+-   .package(url: "https://github.com/pointfreeco/swift-composable-architecture", from: "0.1.0")
++   .package(id: "pointfreeco.swift-composable-architecture", from: "0.1.0")
+]
+```
+
+### Swift package {#swift-package}
+
+If you are working on a Swift package, you can use the `--replace-scm-with-registry` flag to resolve dependencies from the registry if they are available:
+
+```bash
+swift package --replace-scm-with-registry resolve
+```
+
+If you want to ensure that the registry is used every time you resolve dependencies, you will need to update `dependencies` in your `Package.swift` file to use the registry identifier instead of a URL. The registry identifier is always in the form of `{organization}.{repository}`. For example, to use the registry for the `swift-composable-architecture` package, do the following:
+```diff
+dependencies: [
+-   .package(url: "https://github.com/pointfreeco/swift-composable-architecture", from: "0.1.0")
++   .package(id: "pointfreeco.swift-composable-architecture", from: "0.1.0")
+]
+```
+
+## Continuous Integration (CI) {#continuous-integration-ci}
+
+To use the registry on your CI, you need to ensure that you have logged in to the registry by running `tuist registry login` as part of your workflow.
+
+Since the registry credentials are stored in a keychain, you need to set it up as well. Note some CI providers or automation tools like Fastlane already create a temporary keychain or provide a built-in way how to create one. However, you can also create one by creating a custom step with the following code:
+```bash
+TMP_DIRECTORY=$(mktemp -d)
+KEYCHAIN_PATH=$TMP_DIRECTORY/keychain.keychain
+KEYCHAIN_PASSWORD=$(uuidgen)
+security create-keychain -p $KEYCHAIN_PASSWORD $KEYCHAIN_PATH
+security set-keychain-settings -lut 21600 $KEYCHAIN_PATH
+security default-keychain -s $KEYCHAIN_PATH
+security unlock-keychain -p $KEYCHAIN_PASSWORD $KEYCHAIN_PATH
```

Comment:
Not necessary to address here, but some kind of `tuist keychain create --ci` or something along those lines would be awesome.

Reply:
> Not necessary to address here, but some kind of tuist keychain create --ci or something along those lines would be awesome.

I don't think that creating a keychain is a concern that should be handled by Tuist. This rather fits into the idea of Tuist Workflows where you could create a temporary keychain using a library.